qutip/tests/test_superop_reps.py

```diff
@@ -58,295 +59,311 @@
 tol = 1e-9
 
 
-class TestSuperopReps(object):
+
+class TestSuperopReps:
```

Comment:
Is there a reason that you left one function inside this class and moved all the others outside it?  In this case the class isn't needed, because it would just include everything in the file and the filename is enough of a distinction between different testing units, but if we're going to keep it, then all the logically grouped functions should be inside it.

Reply:
The main issue with this one is that if functions go inside the class  `pytest` tries to run them as tests and fails. This is necesary since I am only making  the failing test pytest compatible. But there might be a workaround I do not know of.

Reply:
What do you mean?  `pytest` is the only tool that runs tests in QuTiP.  Also, all the tests run whether or not they're inside the class.

What do you mean by test failures?  I cannot accept a PR that introduces test failures, not matter what your future plans are, and by that I don't just mean hiding tests that don't pass, I mean you mustn't cause performance or logic regressions, and running all the tests ensures that.  A PR is a complete unit of work, it has to be self-contained and make a usable change.  Remember that this is a tool that thousands of people depend on, and any change you make that breaks things can affect users and certainly affects other developers trying to work on the library at the same time - there's a difference in working with a team and working just for yourself.

Reply:
Remember for example this PR #1463 , there we made only some of the tests pytest compatible thus runnable. If we had put  some of the extra test inside a class they would have ran and failed just because they were not written in a pytest amicable format. 
Before this current PR none of the superop_metric tests was being ran, as pytest was not being imported, only unittest.

Reply:
You might want to check you're running the tests correctly.  You can see in the CI logs on GitHub that they are being run, and if you pass the `-v` option when running a single file locally (e.g. `pytest -v test_superop_reps.py`) you'll see clearly that they're all being run locally as well, without any issues.  The only reason that there were issues in #1463 was because it used the old nose-style `yield` format, which `pytest` doesn't support (it uses its more powerful parametrisation instead).

The point about not introducing breaking changes still stands, though - if you thought you were introducing breaking changes, then the correct thing to do would have been to write the test in the same style as the rest of the tests so that they all continued to run, then have a different PR to swap them all to `pytest`.  That way we never compromise our testing, even temporarily.

Reply:
Sorry you are totaly right. The tests are still running if put inside the class. Sorry about that.

Reply:
That's quite ok, no need to apologise!